core/nuxeo-resource.js

```diff
@@ -310,10 +310,14 @@ import './nuxeo-connection.js';
         .execute(options)
         .then((response) =>
           response.text().then((text) => {
-            try {
-              return text ? JSON.parse(text) : {};
-            } catch (e) {
-              return { error: 'Invalid json' };
+            if (options.headers.accept.includes('application/json')) {
```

Comment:
We should check the response headers instead:
```suggestion
            const contentType = response.headers.get('content-type');
            if (contentType && contentType.includes('application/json')) {
```